.github/workflows/bats/sanitize-artifact-name.sh

```diff
@@ -13,29 +13,42 @@ set -o errexit -o nounset -o pipefail
 # allowed to prevent potential problems with downloads on different file
 # systems.
 
-if [[ -z ${1:-} ]]; then
-    echo "usage: $0 LOGDIR"
-    exit 1
+# By default, this script takes a string on standard input and outputs the
+# sanitized string on standard output.  If any positional parameters are given,
+# it instead treats them as file names to (recursively) rename.
+
+sanitize() {
+    local new=$1
+    new=${new//\"/%22}
+    new=${new//:/%3A}
+    new=${new//</%3C}
+    new=${new//>/%3E}
+    new=${new//|/%7C}
+    new=${new//\*/%2A}
+    new=${new//\?/%3F}
+    new=${new//$'\r'/}
+    new=${new//$'\n'/}
+    echo "$new"
+}
+
+if [[ ${#@} -lt 1 ]]; then
```

Comment:
Any reason why you don't want to write `-eq 0`? Can it be less than zero?

Reply:
No, they're equivalent (you can't have a length less than zero). I just used `-lt 1` because that was in my head at the time.